### PR TITLE
feat(new integration): onboarding mailmodo source

### PR DESCRIPTION
**Pull request by rohithkaza**
## Description of the change

> Onboarding Mailmodo Source

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/sources/mailmodo/mapping.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    "sourceKeys": [
+      "triggerData.email",
+      "recipientData.email",
+      "recipientEmail"
+    ],
+    "destKeys": "context.traits.email"
+  },
+  {
+    "sourceKeys": "recipientData",
+    "destKeys": "context.traits.recipientData"
```
Comment (ItsSudip) [inline v0/sources/mailmodo/mapping.json:12]:
```suggestion
    "destKeys": "context.traits"
```
if both `recipientEmail` and `recipientData` are present then remove one email.

File: v0/sources/mailmodo/transform.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    flattenJson(flatEvent)
+  );
+
+  // deleting already mapped fields
+  delete message?.properties["triggerData.email"];
+  delete message?.properties["triggerData.userId"];
+  delete message?.properties["recipientData.email"];
+  delete message?.properties?.recipientEmail;
+  delete message?.properties["triggerData.formSubmissionData.email"];
+  delete message?.properties["triggerData.formSubmissionData.recipientEmail"];
```
Comment (ItsSudip) [inline v0/sources/mailmodo/transform.js:32]:
```suggestion
  message.properties = removeUndefinedAndNullAndEmptyValues(
    flattenJson(flatEvent)
  );
  // deleting already mapped fields
  delete message?.properties.recipientData;
  delete message?.properties.recordedAt;
  delete message?.properties["triggerData.formSubmissionData.recordedAt"];
  delete message?.properties["triggerData.formSubmissionData.recipientData"];
  delete message?.properties["triggerData.email"];
  delete message?.properties["triggerData.userId"];
  delete message?.properties["recipientData.email"];
  delete message?.properties?.recipientEmail;
  delete message?.properties["triggerData.formSubmissionData.email"];
  delete message?.properties["triggerData.formSubmissionData.recipientEmail"];
```
We can put these element inside an array and iterate through that to delete one by one.

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: v0/sources/mailmodo/mapping.json (hunk 1 above)
Comment (rohithkaza) [reply v0/sources/mailmodo/mapping.json]: Done

File: v0/sources/mailmodo/transform.js (hunk 2 above)
Comment (rohithkaza) [reply v0/sources/mailmodo/transform.js]: Done

**Review by ItsSudip — APPROVED** [review]

**Review by sauravlal-rudderstack — APPROVED** [review]

**Pull request merged**